src/modules/paymentProcessor/bridging/abstracts/CrosschainBase_v1.sol

```diff
@@ -0,0 +1,88 @@
+/**
+ * @title   Inverter Template Payment Processor
+ *
+ * @notice  Basic template payment processor used as base for developing new payment processors.
+ *
+ * @dev     This contract is used to showcase a basic setup for a payment processor. The contract showcases the
+ *          following:
+ *          - Inherit from the Module_v1 contract to enable interaction with the Inverter workflow.
+ *          - Use of the IPaymentProcessor_v1 interface to facilitate interaction with a payment client.
+ *          - Implement custom interface which has all the public facing functions, errors, events and structs.
+ *          - Pre-defined layout for all contract functions, modifiers, state variables etc.
+ *          - Use of the ERC165Upgradeable contract to check for interface support.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to our Security Policy
+ *                          at security.inverter.network or email us directly!
+ *
+ * @author  Inverter Network
+ */
+
+abstract contract CrosschainBase_v1 is ICrossChainBase_v1, Module_v1 {
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId_)
+        public
+        view
+        virtual
+        override(Module_v1)
+        returns (bool)
+    {
+        return interfaceId_ == type(ICrossChainBase_v1).interfaceId
+            || super.supportsInterface(interfaceId_);
+    }
+    //--------------------------------------------------------------------------
+    // State
+
+    /// @dev    Payout amount multiplier.
+    uint internal _payoutAmountMultiplier;
+
+    /// @dev    The number of payment orders.
+    uint internal _paymentId;
+
+    /// @dev    The chain id of the current chain.
+    uint internal _chainId;
+
+    //--------------------------------------------------------------------------
+    // Events
+
+    event PaymentProcessed(
+        uint indexed paymentId, address recipient, address token, uint amount
+    );
+
+    //--------------------------------------------------------------------------
+    // Virtual Functions
+
+    /// @notice Execute the cross-chain bridge transfer
+    /// @dev Override this function to implement specific bridge logic
+    /// @param order The payment order containing all necessary transfer details
+    /// @return bridgeData Arbitrary data returned by the bridge implementation
+    function _executeBridgeTransfer(
+        IERC20PaymentClientBase_v1.PaymentOrder memory order,
+        bytes memory executionData
+    ) internal virtual returns (bytes memory) {
+        return bytes("");
+    }
+
+    function getChainId() external view virtual returns (uint) {
+        return _chainId;
+    }
+
+    /// @notice Process payments for a given payment client
+    /// @param client The payment client to process payments for
+    function processPayments(IERC20PaymentClientBase_v1 client)
```

Comment:
This function shouldnt be part of this contract

Reply:
Okay, I removed this but am now curious if this is also true for the state variables and events. Alternatively, what should go in this contract, just `_executeBridgeTransfer`?

Reply:
Yeah thats a good point
I would say that the state in this case can be removed too
The events still fit the purpose and ideally should be used by inherited contract,
Although i am currently not sure if they should be enforced.
Something we could do here, would be to already write the modifiers, that fit to these events,
like a clientIsValid modifier for example.
In general the question to ask is:
Does this functionality fit for the purpose of **this** contract?

Reply:
hmmm okay something to ponder, give a bit. We may need to spar a bit for me to understand this conceptually but I think I grasp the idea. Will make changes and send them over in a bit.